packages/report-server/src/reportBuilder/output/functions/matrix/matrixBuilder.ts

```diff
@@ -58,20 +64,37 @@ export class MatrixBuilder {
     assignColumnSetToMatrixData(columnsInArray);
   }
 
+  private buildPeriod() {
+    const isString = (val: unknown): val is string => typeof val === 'string';
+    const periods = this.rows
+      .map(row => row.period)
+      .filter(isString)
+      .sort();
+
+    if (periods.length > 0) {
+      const earliestPeriod = periods[0];
+      const latestPeriod = periods[periods.length - 1];
+      this.metadata.period = {
+        ...this.metadata.period,
+        reportStart: earliestPeriod,
+        reportEnd: latestPeriod,
+      };
+    }
+  }
```

Comment:
So for now we're just doing this for matrixes, as that's what's required for RN-27. However I feel we really ought to implement output steps for the other tupaia chart types and we can make this logic common across them 👍

Reply:
Definitely agree!